### PR TITLE
Fix auto resign expired certificate

### PR DESCRIPTION
**Pull request by bronzdoc**
# Description:

Related to: https://github.com/rubygems/rubygems/issues/1720

Still need to add a couple of tests...

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by djberg96** [comment]
LGTM, though there are some Travis failures. Not sure what they're about, though.

**Comment by bronzdoc** [comment]
I ran rubygems tests against all the rubies where it's failing and they pass, i think this are bundler specs failing with this change, but still not shure why...

File: lib/rubygems/security/signer.rb [hunk 1]
```diff
@@ -131,22 +139,29 @@ def sign data
...
-    disk_key  =
-      File.read File.join(Gem.default_key_path) rescue nil
+    disk_cert_path = File.join(Gem.default_cert_path)
+    disk_cert = File.read(disk_cert_path) rescue nil
 
-    if disk_key == @key.to_pem and disk_cert == old_cert.to_pem then
-      expiry = old_cert.not_after.strftime '%Y%m%d%H%M%S'
+    disk_key_path = File.join(Gem.default_key_path)
+    disk_key =
+      OpenSSL::PKey::RSA.new(File.read(Gem.default_key_path), @passphrase) rescue nil
```
Comment (bronzdoc) [inline lib/rubygems/security/signer.rb]: Forgot to use `disk_key_path` here

**Comment by djberg96** [comment]
@segiddins Forgot the procedure to merge PR's these days. Can you merge this?

**Comment by segiddins** [comment]
@bundlerbot r+

**Pull request merged**